### DISCUSSION TITLE
The commit history is becoming chaotic

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [pyodide/pyodide#5032] [General]
I used to think that Pyodide maintained a linear commit history, but I've noticed that merge commits are appearing too. No offense intended; I'm just curious about these changes.

**Comment by hoodmane** [comment]
@agriyakhetarpal we prefer to squash merge unless there is a compelling reason to do otherwise. Though personally I don't have a super strong opinion about squash vs merge vs rebase.

**Reply by agriyakhetarpal** [reply]
Oh, sorry, @hoodmane! I definitely prefer squashing as well, but I might have forgotten to do so on some recent occasion and inadvertently created a merge commit. I'll be sure to make a note of this next time. Thanks for pointing it out, @CNSeniorious000.

**Reply by hoodmane** [reply]
I don't really think it's a big deal. And if the commits of the PR are all individually interesting, then merge is perhaps preferable. But most PRs have a bunch of tiny junk commits.

**Reply by agriyakhetarpal** [reply]
Yes, makes sense. I usually find it better to squash because I am lazy to rewrite history :) but I can do it if requested as a part of code review.
